test/test_nn.py

```diff
@@ -7608,5 +7612,34 @@ def __call__(self, input):
     fullname='AdaptiveLogSoftmax'))
 
 
+def disable_tests_not_in_shard(module, shard):
```

Comment:
I would use `load_tests` interface to do this, e.g. https://github.com/pytorch/pytorch/blob/master/test/test_cuda.py#L1882